test/tools/unified-spec-runner/entities.ts

```diff
@@ -195,20 +179,20 @@ export class UnifiedMongoClient extends MongoClient {
   } as const;
 
   constructor(uri: string, description: ClientEntity) {
-    const logCollector = new UnifiedLogCollector(description.observeLogMessages);
+    const logCollector = new UnifiedLogCollector();
     super(uri, {
       monitorCommands: true,
       [Symbol.for('@@mdb.skipPingOnConnect')]: true,
       [Symbol.for('@@mdb.enableMongoLogger')]: true,
-      mongodbLogPath: logCollector,
+      [Symbol.for('@@mdb.internalMongoLoggerConfig')]: {
+        componentSeverities: { default: 'off', ...description.observeLogMessages },
+        logDestination: logCollector
+      },
```

Comment:
It's fine to continue passing the writable into the mongodbLogPath here since that option is for sure going to be part of the final API. The env var config is what we need to be able to configure from here, so I think internalMongoLoggerConfig should point to an object that has `components -> severity`. For now I think we can just pass through observerLogMessages as is, there will need to be some fixing at some point to connect the component names in observerLogMessages to the corresponding env variables.